app/models/spree/product_feed.rb

```diff
@@ -5,17 +5,14 @@ class ProductFeed < Spree::Base
     belongs_to :product_catalog
 
     has_many :images, -> { order(:position) }, as: :viewable, dependent: :destroy, class_name: 'Spree::Image'
+    has_one :overlay_image, dependent: :destroy, class_name: 'Spree::Image', foreign_key: :overlay_image_id
```

Comment:
Nice!! now you can eager load them via associations!!

`Spree::ProductCatalog.find(params[:product_catalog_id]).includes(:overlay_image)` single query and would fetch them all